stubs/jmespath/jmespath/parser.pyi

```diff
@@ -15,5 +18,5 @@ class Parser:
 class ParsedResult:
     expression: str
     parsed: _TreeNode
-    def __init__(self, expression: str, parsed: _TreeNode) -> None: ...
+    def __init__(self, expression: str, parsed: _TreeNodeDict) -> None: ...
```

Comment:
What's the rationale for this change? `dict` invariance? We use `_TreeNode` for all the parameter annotations in `jmespath.visitor`, so I'd be inclined to do so here just for consistency's sake

Reply:
`dict` Invariance yes.
If you think this is better, but want consistency, I'd rather change the other annotations in jmespath as well.

Reply:
Can we maybe leave this one for now, and tackle it in a separate PR? Another contributor added all the `_TreeNode` annotations fairly recently, and I'd like to give them a heads-up before changing them all :)